Windows 365 Gateway IP Lookup/W365 Gateway IP Address Retrieval.ps1

```diff
@@ -0,0 +1,151 @@
+﻿<#
+.COPYRIGHT
+Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+See LICENSE in the project root for license information.
+#>
+
+#v1.0
+
+#output CSV file
+$CSVFile = "c:\temp\W365-Gateways.CSV"
+
+#modules required for this script
+$modules = @("az.accounts",
+    "az.network"
+)
+
+#function to ensure modules are installed and current
+function invoke-modulecheck {
+    foreach ($module in $modules) {
+
+        write-host "Checking module - " $module
+        $graphcurrent = (get-installedmodule -name $module -ErrorAction SilentlyContinue)
+        if ($graphcurrent -eq $null) {
+            write-output "Module is not installed. Installing..." | out-host
+            try {
+                Install-Module -name $module -Force -ErrorAction Stop 
+                Import-Module -name $module -force -ErrorAction Stop 
+
+            }
+            catch {
+                write-output "Failed to install " $module | out-host
+                write-output $_.Exception.Message | out-host
+                Exit
+            }
+        }
+
+        $currentver = (find-module -name $module).version.ToString()
+        $installedver = (get-installedmodule -name $module).version.tostring()
+
+        if ($currentver -gt $installedver) { write-host "There is an update to this module." }
```

Comment:
Would like to understand, if the module won't be updated automatically, why the version be checked here?

Reply:
To inform the user and they can update it themselves. I wanted to avoid having to close all PowerShell sessions as that is required when updating modules.